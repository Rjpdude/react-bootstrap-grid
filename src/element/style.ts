import { mapPropertyStyle } from './util'

export type BaseFlexVal<T> = T | 'start' | 'end' | 'center'

export type GridElementDisplayProperty =
  | 'none'
  | 'inline'
  | 'block'
  | 'inline-block'
  | 'table'
  | 'table-row'
  | 'table-cell'
  | 'flex'
  | 'inline-flex'

export const display = mapPropertyStyle<'display'>(undefined, [
  'none',
  'inline',
  'block',
  'inline-block',
  'table',
  'table-row',
  'table-cell',
  'flex',
  'inline-flex'
])

export type GridElementDirectionProperty =
  | 'row'
  | 'row-reverse'
  | 'column'
  | 'column-reverse'

export const direction = mapPropertyStyle<'direction'>('flexDirection', [
  'row',
  'row-reverse',
  'column',
  'column-reverse'
])

export type GridElementJustifyContentProperty = BaseFlexVal<
  'between' | 'around' | 'evenly'
>

export const justifyContent = mapPropertyStyle<'justifyContent'>(
  (val) => ({
    justifyContent:
      val === 'center'
        ? 'center'
        : val === 'start' || val === 'end'
        ? `flex-${val}`
        : `space-${val}`
  }),
  ['start', 'end', 'center', 'between', 'around', 'evenly']
)

export type GridElementAlignContentProperty = BaseFlexVal<
  'between' | 'around' | 'stretch'
>

export const alignContent = mapPropertyStyle<'alignContent'>(
  (val) => ({
    alignContent:
      val === 'start' || val === 'end'
        ? `flex-${val}`
        : val === 'between' || val === 'around'
        ? `space-${val}`
        : val
  }),
  ['start', 'end', 'center', 'between', 'around', 'stretch']
)

export type GridElementAlignItemsProperty = BaseFlexVal<'baseline' | 'stretch'>

export const alignItems = mapPropertyStyle<'alignItems'>(
  (val) => ({
    alignItems: val === 'start' || val === 'end' ? `flex-${val}` : val
  }),
  ['start', 'end', 'center', 'baseline', 'stretch']
)

export type GridElementAlignSelfProperty = BaseFlexVal<
  'auto' | 'baseline' | 'stretch'
>

export const alignSelf = mapPropertyStyle<'alignSelf'>(
  (val) => ({
    alignSelf: val === 'start' || val === 'end' ? `flex-${val}` : val
  }),
  ['start', 'end', 'center', 'auto', 'baseline', 'stretch']
)

export type GridElementOrderProperty =
  | 'first'
  | 'last'
  | 0
  | 1
  | 2
  | 3
  | 4
  | 5
  | 6
  | 7
  | 8
  | 9
  | 10
  | 11
  | 12

export const order = mapPropertyStyle<'order'>(
  (val) => ({
    order: val === 'first' ? -1 : val === 'last' ? 13 : val
  }),
  ['first', 'last', 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
)

export type GridElementOffsetProperty =
  | 1
  | 2
  | 3
  | 4
  | 5
  | 6
  | 7
  | 8
  | 9
  | 10
  | 11

export const offset = mapPropertyStyle<'offset'>(
  (val) => ({
    marginLeft: (100 / 12) * val
  }),
  [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
)

export type GridElementWrapProperty = boolean | 'reverse'

export const wrap = mapPropertyStyle<'wrap'>(
  (val) => ({
    flexWrap: val === true ? 'wrap' : val === false ? 'nowrap' : 'wrap-reverse'
  }),
  [true, false, 'reverse']
)

export type GridElementGrowthProperty = 0 | 1

export const grow = mapPropertyStyle<'grow'>('flexGrow', [0, 1])

export const shrink = mapPropertyStyle<'shrink'>('flexShrink', [0, 1])

export const fill = mapPropertyStyle<'fill'>(
  (val) =>
    val === true && {
      flex: '1 1 auto'
    },
  [true, false]
)
