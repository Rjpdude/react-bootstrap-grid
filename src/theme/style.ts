import { css, createGlobalStyle } from 'styled-components'

export const Global = css`
  html {
    box-sizing: border-box;
    -ms-overflow-style: scrollbar;
  }

  *,
  *::before,
  *::after {
    box-sizing: inherit;
  }
`

export const GridThemeStyle = createGlobalStyle`
  ${Global}
`
