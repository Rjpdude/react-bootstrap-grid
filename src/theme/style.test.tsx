import * as Style from './style'

test('Global theme style', () => {
  expect(Style.Global).toMatchSnapshot()
})

test('Global theme style component', () => {
  expect(Style.GridThemeStyle).toEqual(expect.any(Function))
})
