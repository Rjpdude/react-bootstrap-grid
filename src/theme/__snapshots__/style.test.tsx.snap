// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global theme style 1`] = `
Array [
  "
  html {
    box-sizing: border-box;
    -ms-overflow-style: scrollbar;
  }

  *,
  *::before,
  *::after {
    box-sizing: inherit;
  }
",
]
`;
